### PR TITLE
Return 0 if there are only notes and no errors

### LINKED ISSUE #10013: Using reveal_type with no errors should return exit code 0
**Bug Report**

When using [pytest-mypy](https://github.com/dbader/pytest-mypy) to test some dynamically determined types (with generics) results in a false-positive error in CI, because mypy's exit code becomes 1 when there are no errors but only `reveal_type` outputs.

**To Reproduce**

Just write an arbitrary Python file that has *no* type errors and put `reveal_type()` to any one of the identifiers.
Run mypy against that file and check the exit code.

I'd like to keep `reveal_type()` in my code, to test a mixture of runtime type validation and static typing results using `typeguard`, generics, and `pytest-mypy` plugin.

**Expected Behavior**

When there are no errors but only some `reveal_type` outputs, `mypy` executable should set the exit code 0.

**Actual Behavior**

It sets the exit code 1, breaking my CI.
image

**Your Environment**

- Mypy version used: 0.800
- Mypy command-line flags and options:
```dosini
[mypy]
ignore_missing_imports = true
mypy_path = stubs,src
namespace_packages = true
explicit_package_bases = true
```
- Python version used: 3.9.1
- Operating system and version: Ubuntu 20.04

### PR DESCRIPTION
**Pull request by ilevkivskyi**
Fixes #10013

See https://github.com/python/mypy/pull/13851#issuecomment-1274345759 for motivation, also this sounds generally reasonable.

### CODE REVIEW

File: mypy/main.py [hunk 1]
```diff
@@ -110,10 +110,10 @@ def main(
         print_memory_profile()
 
     code = 0
-    if messages:
+    n_errors, n_notes, n_files = util.count_stats(messages)
+    if messages and n_notes < len(messages):
```
Comment (hauntsaninja) [inline mypy/main.py:114]: nit: from the impl, it looks like it's not super guaranteed that `n_errors + n_notes == len(messages)`, maybe clearer to phrase as `if messages and n_errors`?

**Review by hauntsaninja — COMMENTED** [review]
Thanks, looks like this also closes https://github.com/python/mypy/issues/10013

**Comment by JukkaL** [comment]
What about notes from `reveal_type()`? I think that these are dangerous, since they generally will fail at runtime. Should they still result in a failed run?

**Comment by hauntsaninja** [comment]
We have at least one feature request for reveal_type to result in exit code 0: https://github.com/python/mypy/issues/10013
We also want exit code 0 when `typing.reveal_type` is used. So I'd say this is fine — we can always make `builtins.reveal_type` emit an error instead of a note.

**Comment by ilevkivskyi** [comment]
Yes, I agree with @hauntsaninja, it is what people probably expecting (also I would propose to deprecate `reveal_type()` in favor of `typing.reveal_type()`).

**Comment by ilevkivskyi** [comment]
I will just go ahead and merge this. It will be easy to undo if someone will complain.

**Pull request merged**